taichi/program/state_flow_graph.cpp

```diff
@@ -200,21 +273,19 @@ bool StateFlowGraph::optimize_listgen() {
     delete_nodes(nodes_to_delete);
     // Note: DO NOT topo sort the nodes here. Node deletion destroys order
     // independency.
-    auto tasks = extract(/*sort=*/false);
-    for (auto &task : tasks) {
-      insert_task(task);
-    }
+    rebuild_graph(/*sort=*/false);
   }
 
   return modified;
 }
 
 std::pair<std::vector<bit::Bitset>, std::vector<bit::Bitset>>
-StateFlowGraph::compute_transitive_closure() {
+StateFlowGraph::compute_transitive_closure(int begin, int end) {
```

Comment:
If we don't reset `pending_node_id` in this function, the index `pending_node_id` should subtract by `begin`, because the index range of `Bitset` is `[0, end - begin)` instead of `[begin, end)`.